### PR TITLE
Attempt to run Netlify only when there is both a pull request against…

### PR DESCRIPTION
**Pull request by stevepiercy**
… master and a file has changed in the `docs` directory

We just got a notification from Netlify:

> Your build minutes usage on your team (Plone) has reached 50% of the current allowance in your billing cycle from February 23 to March 23.

Roughly two-thirds of the build minutes are from Volto.

Site | Builds | Average time | Total time
-- | -- | -- | --
volto | 203 | 1m 56s | 390m
6-dev-docs-plone-org | 43 | 2m 29s | 106m
plone-training | 27 | 2m 42s | 73m
plone-restapi | 5 | 9s | 1m

Most of the PRs are not related to docs. @ksuess said that it is possible to build on Netlify only on PRs that include changes to files in `docs/`. I found this [YAML syntax for workflows documentation](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore) that confirms it.

This PR attempts to configure it to do that thing.

### CODE REVIEW

**Comment by stevepiercy** [comment]
Hm, it appears that I cannot test this in a PR. And I do not want to directly to `master` without checking with @sneridagh, or handing it off to a Volto maintainer.

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -3,6 +3,8 @@ on:
   push:
     branches:
       - master
+    paths:
+      - 'docs/**'
     # Build pull requests
     pull_request:
```
Comment (davisagli) [inline .github/workflows/docs.yml:9]: This line is at the wrong indentation (pull_request is another event type like push). If your intent is to run this for commits on master that touch docs AND for pull requests that touch docs, then I think you want:
```suggestion
  # Build pull requests
  pull_request:
    paths:
      - 'docs/**'
```

Comment (stevepiercy) [reply .github/workflows/docs.yml]: Yes, that was my intent. Thank you! I've committed your suggestion.

**Comment by stevepiercy** [comment]
@sneridagh I lack permission to merge to `master`. Were you expecting me to merge this PR, or did you want to do it and fiddle with the configuration by pushing changes to `master` directly in case it does not do what we want?

**Pull request merged**